src/domain/core/src/entities/dataset.rs

```diff
@@ -255,3 +256,30 @@ impl From<CheckpointRef> for Option<OwnedFile> {
         }
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Copy, Clone, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub enum DatasetVisibility {
+    #[default]
+    Private,
+    PubliclyAvailable,
```

Comment:
Why `PubliclyAvailable` instead of a concise `Public`? I don't see any ambiguity that a longer name would help to resolve.

Also you could add serde tags to allow deserializing these fields from all variations `Pulbic / public / Private / private` and reuse these in CLI.

When we move towards declarative configurations I wouldn't want to spell out `PubliclyAvailable` in the configs.

Reply:
> Why PubliclyAvailable instead of a concise Public? I don't see any ambiguity that a longer name would help to resolve.

Changed -- Done

> Also you could add serde tags to allow deserializing these fields from all variations Pulbic / public / Private / private and reuse these in CLI

Reused -- Done